components/prism-bash.js

```diff
@@ -19,66 +29,168 @@
 					punctuation: /\(\(?|\)\)?|,|;/
 				}
 			},
-			// Command Substitution
+			// [1]: Command Substitution
 			{
-				pattern: /\$\([^)]+\)|`[^`]+`/,
+				pattern: /\$\((?:\([^)]+\)|[^()])+\)|`[^`]+`/,
 				greedy: true,
 				inside: {
 					variable: /^\$\(|^`|\)$|`$/
 				}
 			},
-			/\$(?:[\w#?*!@]+|\{[^}]+\})/i
-		]
+			// [2]: Brace expansion
+			{
+				pattern: /\$\{[^}]+\}/,
+				greedy: true,
+				inside: {
+					operator: /:[-=?+]?|[!\/]|##?|%%?|\^\^?|,,?/,
+					punctuation: /[\[\]]/,
+					environment: {
+						pattern: new RegExp("(\\{)(?:" + envVarList + ")(?=$|\\}|\\W)"),
+						lookbehind: true,
+						alias: 'constant'
+					}
+				}
+			},
+			/\$(?:\w+|[#?*!@$])/
+		],
+		// man echo / printf, + \"
+		entity: /\\([abceEfnrtv\\"]|O?[0-7]{1,3}|x[0-9a-fA-F]{1,2}|u[0-9a-fA-F]{4}|U[0-9a-fA-F]{8})/
```

Comment:
1. Please make the group non-capturing.
2. `[abceEfnrtv\\"]` looks like black magic to me. Could you please explain this to someone who doesn't know bash?

Reply:
Now that you mention it… These are the characters than can have special meanings when escaped within a string. Well, it ultimately depends on the command that receives the string, but as in 99.9 % of cases the meaning will indeed be special, I thought it would be a shame not to highlight such sequences. This subset is some sort of union of what `echo` and `printf` support. The `"` is for escaped quotes, because string that contain quotation marks tend to get unreadable fast in Bash.

I'll make the existing comment more… human-readable, I guess.

Reply:
> I'll make the existing comment more… human-readable, I guess.

You don't have to. It's clear to me now. Thank you.